### ISSUE TITLE
Request: Retractable Panels

### ISSUE THREAD

**Issue by andremiles1** [BEEmod/BEE2-items#1650]
Angled/Glass Panels have these arms to stand it, and sometimes this is a problem. Players can easily jump into it too.

![untitled-1](https://cloud.githubusercontent.com/assets/26036746/23573847/dce2ef74-0057-11e7-93c8-1e2379f2abda.jpg)

![untitled-2](https://cloud.githubusercontent.com/assets/26036746/23573850/e0d951ae-0057-11e7-9158-0aac40840a1a.jpg)

Doors, when placed alone or with windows on it sides, it get this weird look.

![untitled-3](https://cloud.githubusercontent.com/assets/26036746/23573859/f2eae8c6-0057-11e7-8864-129ee9383814.jpg)

So, I suggest a new type of panel that retract to the ground or wall they're placed. Like a door, but instead of "cutting in half" to open, the panel will go full to the side. I can't make a drawing of this right now, so I hope I'm being clear.
I imagine they making the same sound as Piston Platforms.

**Comment by TeamSpen210** [comment]
You mean like this?
![image](https://cloud.githubusercontent.com/assets/8453732/23574851/59f236ce-00cf-11e7-90db-eb0015ad80d1.png)

**Comment by andremiles1** [comment]
Almost. They go full inside the ground/wall to one side.

**Comment by andremiles1** [comment]
Imagine this is the Retractable Panel active.
![untitled-1](https://cloud.githubusercontent.com/assets/26036746/23575422/7c1b96c4-006a-11e7-9fe3-7bcefddb1faa.jpg)

As you inactive it, it goes down like this
![untitled-2](https://cloud.githubusercontent.com/assets/26036746/23575423/8590ed12-006a-11e7-99a6-48838a7d28e4.jpg)

Or inactive when placed on a wall, like this
![untitled-3](https://cloud.githubusercontent.com/assets/26036746/23575425/8d853640-006a-11e7-9302-b11004cec27e.jpg)

**Comment by TeamSpen210** [comment]
Ah, that's a good idea as well.

**Comment by Luke18033** [comment]
This could use the Portal 1 vertical security door model.

**Comment by TheDarkBomber** [comment]
No, that would be a different item.

**Comment by Luke18033** [comment]
What would it be? The P1 vertical door I assume would be one of those "turret doors" that appear in BTS areas, since it is most similar to the P2 vertical door:
![20170304083939_1](https://cloud.githubusercontent.com/assets/19228201/23580412/3d9c6bb6-00b6-11e7-80a7-96a74f962c2c.jpg)
This item should be the other style of vertical door, like this:
![20170304084558_1](https://cloud.githubusercontent.com/assets/19228201/23580604/ad5c5454-00b9-11e7-9a32-b7d1814b6229.jpg)

**Comment by TheDarkBomber** [comment]
That turret door should be the styled VSD for P1 BTS. The other should be the styled VSD for P1.

**Comment by Luke18033** [comment]
No. Normal P1 should use a recolored version of the BTS P1 door.

**Comment by Luke18033** [comment]
Items need to be similar between styles to prevent breaking maps.

**Comment by TheDarkBomber** [comment]
I don't agree with you, that door is normally used as a VSD in Portal.

**Comment by Luke18033** [comment]
It retracts differently though. See #1648 - items need to be similar between styles, to prevent breaking maps. A recolored version of the first door should be used. It would be white with a blue light, like the normal doors. The second door would be another item.

**Comment by Endermage77** [comment]
Maybe it could be an Alternative item variant if it were to be the P1 Vertical Security Door?

Though personally, I would prefer the solid panel appearance as shown.

**Comment by andremiles1** [comment]
I'm glad everyone liked my idea. Hope it gets implemented!

**Comment by Luke18033** [comment]
@TheDarkBomber - The second door would probably be the main variant, with a recolored version of the first as an alternate version.

**Comment by Endermage77** [comment]
Come to think of it, why is the P1 exit door a Vertical Security Door?

**Comment by Luke18033** [comment]
That's only when corridor 4 is selected. It should really be a stylevar option so it can apply to any corridor.
